lib/CacheManager.js

```diff
@@ -1,56 +1,359 @@
-};
-
-CacheManager.prototype._debug = function _debug(msg){
-  if (this.debug) {
-    console.log(msg)
-  }
-};
-
-CacheManager.prototype.setObject = function setObject(key, value) {
-  var newValue,
-      storageSpace = this.storageSpace,
-      objSerializer = this.objectSerializer;
+define(['GarbageCollector', 'ObjectSerializer', 'SetSerializer'], function(garbageCollector, objectSerializer, setSerializer){
 
-  newValue = objSerializer.serialize(value);
-  storageSpace.setItem(key, newValue)
-};
+  /*
+   *   CacheManager() is responsible for implementing the public API methods of cache-man.
+   *
+   *   These public API methods are aimed at providing an easy to use way of serializing
+   *   and deserializing data that is cached in the window, either sessionStorage or
+   *   localStorage.
+   *
+   *
+   * */
+  var CacheManager = function CacheManager(storageSpace) {
+    this.storageSpace = storageSpace;
+    this.collector = new garbageCollector;
+    this.objectSerializer = new objectSerializer;
+    this.setSerializer = new setSerializer;
 
-CacheManager.prototype.setCategoricalSets = function setItems(key, setsToSerialize) {
-  var newValues,
-      storageSpace = this.storageSpace,
+
+    this.collectionQueue = [];
+  };
+
+  /*
+   *  @key        String
+   *  @value      String or Object
+   *  @namespace  String
+   *
+   *  Public API Method -- Setter for Individual Entities
+   *
+   *  setObject() is responsible for setting a value, with its key, in the proper namespace
+   *  of the cache. setObject() calls two private methods, _prepareObjectForCache() which serializes
+   *  the value through JSON protocol for caching, and _retryCachingObject() which recursively
+   *  tries to add the object to the cache.
+   *
+   *
+   * */
+  CacheManager.prototype.setObject = function setObject(key, value, namespace) {
+    var data;
+
+    //throw error if key is not a string
+    if ((key && namespace) === typeof String) {
+      throw ('Error. Key and namespace are not strings')
+    }
+
+    //if @namespace param isn't present, exit early
+    data = this._prepareObjectForCache(key, value, namespace);
+
+
+    // if there is space in the local cache, keep retrying
+    this._retryCachingObject(data, namespace);
+  };
+
+  /*
+   *  @key        String
+   *  @value      String or Object
+   *  @namespace  String
+   *
+   *  Private API Method
+   *
+   *  _prepareObjectForCache() is responsible for serializing the value through JSON protocol, in
+   *  preparation for caching. The method returns the actual key-value pair object to set, as well
+   *  as a reference to the item in the cache through a key-value pair of the namespace and the unique
+   *  key of the the item to be cached.
+   *
+   * */
+  CacheManager.prototype._prepareObjectForCache = function _prepareObjectForCache(key, value, namespace) {
+    var newValue,
+      objSerializer = this.objectSerializer,
+      objToSet = {},
+      cachedObj = {};
```

Comment:
Please use separate `var` statements for easier refactoring.  Also, declarations without default values should go last